src/boot/efi/util.c

```diff
@@ -110,13 +109,7 @@ EFI_STATUS efivar_get(const EFI_GUID *vendor, const char16_t *name, char16_t **r
                 return EFI_SUCCESS;
         }
 
-        /* Make sure a terminating NUL is available at the end */
-        val = xmalloc(size + sizeof(char16_t));
-
-        memcpy(val, buf, size);
-        val[size / sizeof(char16_t) - 1] = 0; /* NUL terminate */
-
-        *ret = val;
+        *ret = xstrndup16(buf, size / sizeof(char16_t));
```

Comment:
hmm, so static checkers in userspace generally assume that strndup() is invoked on valid NUL terminated strings only, and if not bad things happen. Because of that we added to userspace a helper memdup_suffix0() which works on any kind of memory. Even though static analyzers probably won't recognize xstrndup16() as a flavour of strndup() I still think we probably should do things similar in efi mode too